From ab49ad4bb7e0dd239871a664ca9584c07459348a Mon Sep 17 00:00:00 2001
From: Dong Aisheng <aisheng.dong@nxp.com>
Date: Thu, 7 Sep 2017 12:40:38 +0800
Subject: [PATCH 3318/5242] MLK-17491-23 clk: imx: clk-pllv4: fix wrong return
 of clk_pllv4_is_enabled

commit  442a0842c99e493bf45b31c9642340a7bc3d07d3 from
https://source.codeaurora.org/external/imx/linux-imx.git

PLL_EN bit set means pll enabled, not disabled.

Fixes: 78ef764871d6 ("MLK-13441-5 ARM: imx: add new clk types")
Cc: Anson Huang <Anson.Huang@nxp.com>
Reviewed-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Dong Aisheng <aisheng.dong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-pllv4.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/clk/imx/clk-pllv4.c b/drivers/clk/imx/clk-pllv4.c
index 27f4280..1d2ab54 100644
--- a/drivers/clk/imx/clk-pllv4.c
+++ b/drivers/clk/imx/clk-pllv4.c
@@ -131,9 +131,9 @@ static int clk_pllv4_is_enabled(struct clk_hw *hw)
 	struct clk_pllv4 *pll = to_clk_pllv4(hw);
 
 	if (readl_relaxed(pll->base) & PLL_EN)
-		return 0;
+		return 1;
 
-	return 1;
+	return 0;
 }
 
 static const struct clk_ops clk_pllv4_ops = {
-- 
1.7.9.5

